pre-commit.template

```diff
@@ -0,0 +1,123 @@
+#!/bin/sh
+#
+# PreReq's
+#  - This requires that a current level of Eclipse is defined on your machine 
+
+# This hook will run the eclipse code formatter before any commit.
+# This will format Java code files that are part of the commit.
+# The java code will be formatted based on the preferences defined
+# in the java.prefs file provided in the base directory of this project.
+#
+# Update the following items below then save this file in 
+# .git/hooks/pre-commit 
+# be sure you have execute authority for this file. 
+
+# set this to the location of your Eclipse installtion
+# Mac OSX Example: /Users/mac/EclipseInstalls/EclipseO2/EClipse.app/Contents/MacOS
```

Comment:
Might be nice to have an example for Windows and Linux too.

Reply:
Good Point, which has me thinking. This will obviously only work on linux/Mac/cygwin. Wondering if I should indicate this won't run in windows unless a unix environment is provided
(Cygwin) Which begs the question do we need a windows only version of this functionality?